firstvoices/backend/resources/characters.py

```diff
@@ -38,3 +43,27 @@ class Meta:
 class IgnoredCharacterResource(SiteContentResource):
     class Meta:
         model = IgnoredCharacter
+
+
+class AlphabetConfusablesResource(SiteContentResource):
+    "Convert rows to a JSON list and save to the site Alphabet."
+
+    class Meta:
+        model = Alphabet
+
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        if not dry_run:
+            alphabet, _ = Alphabet.objects.get_or_create(site_id=dataset["site"][0])
+
+            list_of_mappers = []
+            for in_form, out_form in zip(dataset["in_form"], dataset["out_form"]):
+                mapper = {"in": in_form, "out": out_form}
+                list_of_mappers.append(mapper)
+
+            alphabet.input_to_canonical_map = list_of_mappers
+            alphabet.save()
+        return super().before_import(dataset, using_transactions, dry_run, **kwargs)
+
+    def skip_row(self, instance, original, row, import_validation_errors=None):
+        "Never import individual rows."
```

Comment:
[minor] Is this meant to be a comment? It seems like just a string without an assignment.